### PR TITLE
Adding syncthing api unittests.

### PR DESCRIPTION
**Pull request by jmacelroy**
Added a test for the proper insertion of headers by the syncthing client roundtripper and the proper initialization of the request timeout for the http client.

### CODE REVIEW

File: pkg/syncthing/api.go [hunk 1]
```diff
@@ -30,8 +30,15 @@ type addAPIKeyTransport struct {
 	T http.RoundTripper
 }
 
+const (
+	APIKeyHeader      = "X-Api-Key"
+	APIKeyHeaderValue = "cnd"
+)
+
+// RoundTrip implements the http.RoundTripper interface and is used to add the
+// desired request headers to http requests.
```
Comment (andreafalzetti) [inline pkg/syncthing/api.go:39]: What's the context behind this change? Where does the value `cdn` come from?

**Review by jLopezbarb — APPROVED** [review]
LGTM!! Thanks a lot @jmacelroy

File: pkg/syncthing/api.go (hunk 1 above)
Comment (andreafalzetti) [reply pkg/syncthing/api.go:39]: I've seen the same API KEY being set when we init syncthing, so it makes more sense now 👍

**Review by andreafalzetti — APPROVED** [review]

File: pkg/syncthing/api.go (hunk 1 above)
Comment (jmacelroy) [reply pkg/syncthing/api.go:39]: fwiw I do not know the answer to your question on , "why we are sending this header?" The approach was to keep the code doing what it was doing before, but with a test.

I did the overall thing just to go through the flow of a unittest in okteto/okteto while investigating making changes for our approach to test coverage.

**Pull request merged**